clientupdate/clientupdate.go

```diff
@@ -259,13 +266,16 @@ func Update(args Arguments) error {
 }
 
 func (up *Updater) confirm(ver string) bool {
-	switch cmpver.Compare(version.Short(), ver) {
-	case 0:
-		up.Logf("already running %v version %v; no update needed", up.Track, ver)
-		return false
-	case 1:
-		up.Logf("installed %v version %v is newer than the latest available version %v; no update needed", up.Track, version.Short(), ver)
-		return false
+	// Only check version when we're not switching tracks.
+	if up.Track == "" || up.Track == CurrentTrack {
```

Comment:
Is "" still possible given the new logic above?

Reply:
I don't think it's possible, but I left it just to be extra defensive.
We don't allow setting the Track on several platforms, which makes me worried that an empty string will percolate somehow.
https://github.com/tailscale/tailscale/blob/fbb4e5248a6d87cf85fc0c2328609111261b6f37/cmd/tailscale/cli/update.go#L29-L43